L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -13,8 +14,20 @@ HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
     return TC_id.cell(); //scintillator
   } else {
     int TC_wafer = TC_id.wafer();
-    int TC_12th = ( TC_id.cell() & kSplit_ );
-    return TC_wafer<<kWafer_offset_ | TC_12th;
+    int split_12th = 0x3a;
+    int split_3rd  = 0x30;
+    int thickness = triggerTools_.thicknessIndex(detid);
+
+    int TC_12th = ( TC_id.cell() & split_12th );
+    int TC_3rd  = ( TC_id.cell() & split_3rd );
+
+    int TC_split = TC_12th;
+
+    if (stcSize_.at(thickness) == 16)
+      TC_split = TC_3rd;
+
+    int wafer_offset = 6;
```

Comment:
This is a regression. Please include the modifications that are in CMSSW 10_5_0_pre1.